src/abieos.cpp

```diff
@@ -111,7 +111,9 @@ extern "C" abieos_bool abieos_set_abi(abieos_context* context, uint64_t contract
             return set_error(context, std::move(error));
         abieos::abi c;
         convert(def, c);
-        context->contracts.insert({name{contract}, std::move(c)});
+        const name key {contract};
+        context->contracts.erase(key);
+        context->contracts.insert({key, std::move(c)});
```

Comment:
Should be able to use:  `context->contracts.insert_or_assign(name{contract}, std::move(c));`